src/main/java/org/apache/mesos/offer/ResourceRequirement.java

```diff
@@ -15,12 +20,103 @@
  * or an already determined value.
  */
 public class ResourceRequirement {
+    /* keys for log */
+    public static final String ENV_KEY = "Environment_Name";
+    public static final String VIP_KEY = "VIP_KEY";
+    public static final String VIP_VALUE = "VIP_VALUE";
+
+
     private MesosResource mesosResource;
     private DiskInfo diskInfo;
 
+    private Optional<String> envName = Optional.empty();
+    private Optional<Label> vipLabel = Optional.empty();
+
     public ResourceRequirement(Resource resource) {
         this.mesosResource = new MesosResource(resource);
         this.diskInfo = getDiskInfo();
+        ResourceUtils.setEnvName(this);
+        ResourceUtils.setVIPLabel(this);
+
+    }
+
+    public boolean hasEnvName() {
+        if (envName.isPresent()) {
+            return true;
+        }
+        return false;
+    }
+
+    public boolean hasVIPLabel() {
+        return vipLabel.isPresent();
+    }
+
+
+    public String getEnvName() {
+        if (hasEnvName()) {
+            return envName.get();
+        }
+        return null;
+    }
+
+    public Label getVIPLabel() {
+        if (hasVIPLabel()) {
+            return vipLabel.get();
+        }
+        return null;
+    }
+
+
+    public void setEnvName(String envName) {
+        this.envName = Optional.of(envName);
+        if (!this.envName.isPresent()) {
```

Comment:
How could this condition ever occur?  If `envName` is null?  You can't create an `Optional.of(null)`.  It will throw.